tests/schema/tests.py

```diff
@@ -3510,6 +3510,47 @@ def test_unique_constraint_nulls_distinct_unsupported(self):
             self.assertIsNone(editor.add_constraint(Author, constraint))
             self.assertIsNone(editor.remove_constraint(Author, constraint))
 
+    @skipUnlessDBFeature("supports_nulls_distinct_unique_constraints")
+    def test_create_unique_nulls_distinct(self):
+        """
+        Regression test for #35002.
+        """
+        with connection.schema_editor() as editor:
+            editor.create_model(Author)
+        constraint = UniqueConstraint(
+            fields=("height", "weight"), name="constraint", nulls_distinct=False
+        )
+        with connection.schema_editor() as editor:
+            editor.add_constraint(Author, constraint)
+        Author.objects.create(name="", height=None, weight=None)
+        Author.objects.create(name="", height=1, weight=None)
+        Author.objects.create(name="", height=None, weight=1)
+        with self.assertRaises(IntegrityError):
+            Author.objects.create(name="", height=None, weight=None)
+        with self.assertRaises(IntegrityError):
+            Author.objects.create(name="", height=1, weight=None)
+        with self.assertRaises(IntegrityError):
+            Author.objects.create(name="", height=None, weight=1)
+        with connection.schema_editor() as editor:
+            editor.remove_constraint(Author, constraint)
+        constraints = self.get_constraints(Author._meta.db_table)
+        self.assertNotIn(constraint.name, constraints)
+
+    @skipIfDBFeature("supports_nulls_distinct_unique_constraints")
+    def test_create_unique_nulls_distinct_unsupported(self):
+        """
+        Supplement to regression test for #35002.
+        """
+        with connection.schema_editor() as editor:
+            editor.create_model(Author)
+        constraint = UniqueConstraint(
+            fields=("height", "weight"), name="constraint", nulls_distinct=False
+        )
+        with connection.schema_editor() as editor:
+            self.assertIsNone(editor.add_constraint(Author, constraint))
+        constraints = self.get_constraints(Author._meta.db_table)
+        self.assertNotIn(constraint.name, constraints)
+
```

Comment:
I'm not sure this is needed. Happy to remove if preferred; I added it for completeness.

Note that the assertion is different than in the corresponding part of `test_unique_constraint_nulls_distinct_unsupported`, because adding/removing the constraint actually *does* trigger queries. They are no-op:
```
AssertionError: 8 != 0 : 8 queries executed, 0 expected
Captured queries were:
1. CREATE TABLE "new__schema_author" ("id" integer NOT NULL PRIMARY KEY AUTOINCREMENT, "name" varchar(255) NOT NULL, "height" integer unsigned NULL CHECK ("height" >= 0), "weight" integer NULL, "uuid" char(32) NULL)
2. INSERT INTO "new__schema_author" ("id", "name", "height", "weight", "uuid") SELECT "id", "name", "height", "weight", "uuid" FROM "schema_author"
3. DROP TABLE "schema_author"
4. ALTER TABLE "new__schema_author" RENAME TO "schema_author"
5. CREATE TABLE "new__schema_author" ("id" integer NOT NULL PRIMARY KEY AUTOINCREMENT, "name" varchar(255) NOT NULL, "height" integer unsigned NULL CHECK ("height" >= 0), "weight" integer NULL, "uuid" char(32) NULL)
6. INSERT INTO "new__schema_author" ("id", "name", "height", "weight", "uuid") SELECT "id", "name", "height", "weight", "uuid" FROM "schema_author"
7. DROP TABLE "schema_author"
8. ALTER TABLE "new__schema_author" RENAME TO "schema_author"
```
Ditto for removing the constraint.

Not sure whether it's worth doing anything about that (but probably not in this PR anyway).

Reply:
yeah I think we can do without!